### PR TITLE
Revamped Push Block

### PR DESCRIPTION
**Pull request by awjuliani**
* Adds new revamped Push Block environment.
* Adds "Shared Assets" folder to Examples sub-directory.

### CODE REVIEW

File: unity-environment/Assets/ML-Agents/Examples/PushBlock/Scripts/GoalDetect.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿//Detect when the orange block has touched the goal. 
+//Detect when the orange block has touched an obstacle. 
+//Put this script onto the orange block. There's nothing you need to set in the editor.
+//Make sure the goal is tagged with "goal" in the editor.
+
+using UnityEngine;
+
+public class GoalDetect : MonoBehaviour {
+	[HideInInspector]
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Examples/PushBlock/Scripts/GoalDetect.cs]: save tabs as spaces to fix indentation problems (in several files)

File: unity-environment/ProjectSettings/ProjectVersion.txt [hunk 2]
```diff
@@ -1 +1 @@
-m_EditorVersion: 2018.1.0b6
+m_EditorVersion: 2018.1.0b9
```
Comment (mmattar) [inline unity-environment/ProjectSettings/ProjectVersion.txt]: Would it make sense to ignore this file from git?

File: unity-environment/Assets/ML-Agents/Examples/Shared Assets/Scripts/SliderValueUI.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+	public Slider sliderToUse;
+	public Text textUI;
+
+	// Use this for initialization
+	void Start () {
+		textUI.text = sliderToUse.value.ToString();
+	}
+	
+	// // Updat
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Examples/Shared Assets/Scripts/SliderValueUI.cs]: remove

File: unity-environment/Assets/ML-Agents/Examples/Shared Assets/Scripts/SliderValueUI.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	
+	public void UpdateSliderValueUI(float val)
+	{
+		// textUI.text = val.ToString();
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Examples/Shared Assets/Scripts/SliderValueUI.cs]: remove

File: unity-environment/Assets/ML-Agents/Examples/Shared Assets/Scripts/MLAgentsHelpers.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public static class MLAgentsHelpers{
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Examples/Shared Assets/Scripts/MLAgentsHelpers.cs]: move brace to newline

File: unity-environment/Assets/ML-Agents/Examples/Shared Assets/Scripts/MLAgentsHelpers.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        agent.AddVectorObs(t.rotation.eulerAngles.x / 180.0f - 1.0f);
+        agent.AddVectorObs(t.rotation.eulerAngles.y / 180.0f - 1.0f);
+        agent.AddVectorObs(t.rotation.eulerAngles.z / 180.0f - 1.0f);
+    }
+
+    /// <summary>
+    /// We can only collect floats in CollecState,
+    /// so in some cases it is helpful to convert bools to floats.
+    /// </summary>
+    public static float ConvertBoolToFloat(bool b)
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Examples/Shared Assets/Scripts/MLAgentsHelpers.cs]: does Convert.ToSingle(b) replace this method?

Comment (awjuliani) [reply unity-environment/Assets/ML-Agents/Examples/Shared Assets/Scripts/MLAgentsHelpers.cs]: Yes. Removing.

File: unity-environment/Assets/ML-Agents/Examples/PushBlock/Scripts/GoalDetect.cs (hunk 1 above)
Comment (awjuliani) [reply unity-environment/Assets/ML-Agents/Examples/PushBlock/Scripts/GoalDetect.cs]: Reformatted the push block files w/ spaces.

File: unity-environment/ProjectSettings/ProjectVersion.txt (hunk 2 above)
Comment (awjuliani) [reply unity-environment/ProjectSettings/ProjectVersion.txt]: It is actually needed for Unity when importing a project into a new version of the Editor.

**Pull request merged**